tools/mtouch/mtouch.cs

```diff
@@ -394,9 +395,11 @@ public static string GetAotArguments (Application app, string filename, Abi abi,
 			args.Append ("-ios,");
 			args.Append ("data-outfile=").Append (StringUtils.Quote (dataFile)).Append (",");
 			args.Append (app.AotArguments);
-			if (llvm_only)
+			if (llvm_only && interp) {
+				args.Append ("interp,llvmonly,");
+			} else if (llvm_only) {
 				args.Append ("llvmonly,");
-			else if (interp) {
+			} else if (interp) {
 				if (fname != "mscorlib.dll")
 					throw ErrorHelper.CreateError (99, $"Internal error: can only enable the interpreter for mscorlib.dll when AOT-compiling assemblies (tried to interpret {fname}). Please file an issue at https://github.com/xamarin/xamarin-macios/issues/new.");
 				args.Append ("interp,");
```

Comment:
adding the same value in multiple place makes things harder to understand (and easier to miss when updating).
I think you can have a single `if (interp) { args.Append ("interp,"); }`